src/ert/_c_wrappers/analysis/analysis_module.py

```diff
@@ -172,12 +174,15 @@ def set_var(self, var_name: str, value: Union[float, int, bool, str]):
                     var["value"] = new_value
 
             except ValueError:
-                raise ValueError(
-                    f"Variable {var_name} expected type {var['type']}"
-                    f" received value `{value}` of type `{type(value)}`"
+                raise ConfigValidationError(
+                    f"Variable {var_name} with value `{value}` has incorrect type."
+                    f" Expected type {var['type']} but received value `{value}` of type"
+                    f" `{type(value)}`"
                 )
         else:
-            raise KeyError(f"Variable {var_name} not found in module")
+            raise ConfigValidationError(
+                f"Variable {var_name} not found in {self.name} analysis module"
```

Comment:
The proposal was to use f-strings with "!r" format specificer.

Reply:
right, sorry, i'm a bit late to the discussion and didn't notice it here - cool, I didn't know about `!r`. it helps with simple string values:
```
>>> x = "foo"
>>> y = f"something went wrong with {x!r} when doing bar"
>>> print(y)
something went wrong with 'foo' when doing bar
```
but once we for instance have quotation marks involved, things become meh:
```
>>> x = "'foo'"
>>> y = f"something went wrong with {x!r} when doing bar"
>>> print(y)
something went wrong with 'foo' when doing bar
```
the original extra quotes in the value of x are swallowed.

While this will only be relevant in a few cases, in my experience of sometimes having to debug exactly that kind of "variable substitution here or there swallowed some quotation marks" i prefer using the backticks, as in

```
>>> y = f"something went wrong with `{x}` when doing bar"
```

as backticks are extremely rarely used in strings (in my experience), and i suppose the effort is comparable to using `!r`

Reply:
I'm a bit torn on this matter, but have to ask, what python version do you run, and can you re-check that last example?

```
Python 3.8.10 (default, Nov 14 2022, 12:59:47)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> x = "'foo'"
>>> y = f"something went wrong with {x!r} when doing bar"
>>> print (y)
something went wrong with "'foo'" when doing bar
>>>
```

My ticks were not swallowed like yours were?

Reply:
let me check maybe i did the dumbs